pkg/server/debug/goroutineui/dump.go

```diff
@@ -42,6 +43,9 @@ func (d Dump) SortCountDesc() {
 	if d.err != nil {
 		return
 	}
+	if d.agg == nil {
```

Comment:
Is it possible that in `newDumpFromBytes` above, the err being returned could be nil? It wouldn't be EOF then and we'd never write the `agg` field. Unsure if that's even possible though.

Reply:
Tough to tell from the library code, but it's good to guard against. It seems like that's what probably happened to cause the panic. `err` was `nil` so we leave `agg` set to `nil` which leads to panic when trying to read `d.agg.Buckets`.

Are you implying that I should modify `newDumpFromBytes`?

Reply:
I'm not sure what state the returned snapshot would be if err was nil so I think keeping the nil agg checks makes the most sense. Was just trying to throw out an explanation for how we got to this state :)